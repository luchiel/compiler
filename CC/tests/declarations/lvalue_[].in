void main()
{
int a[2];
a[0] = 1;
}