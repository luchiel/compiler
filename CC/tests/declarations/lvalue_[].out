type <int>
type <float>
type <void>
type <int*>
	pointer to <int>
type <void*>
	pointer to <void>
variable NULL of type <void*> initialized by
	{int 0}
		type <int>
function printf
	returns type <int>
	arguments
		<no args>
function main
	returns type <void>
	arguments
		<no args>
	{{}}
		type <0>
			array [
				{int 2}
					type <int>
			] of <int>
		variable a of type <0>
	| 
	+-{;}
	  | 
	  +-{=}
				type <int>
	    | 
	    +-{[]}
					type <int>
	    | | 
	    | +-{a}
						type <0>
							array [
								{int 2}
									type <int>
							] of <int>
	    | | 
	    | +-{int 0}
						type <int>
	    | 
	    +-{int 1}
					type <int>
